pymodbus/client/mixin.py

```diff
@@ -508,7 +508,7 @@ def read_device_information(
     # ------------------
 
     class DATATYPE(Enum):
-        """Datatype enum for convert_* calls."""
+        """Datatype name and number of bytes, used for convert_* calls."""
```

Comment:
The enum is the name, the first part of the tuple is format code for struct.
```suggestion
        """Datatype enum used in convert_* calls."""
```
I think it is important to state that its an enum for the user, the content is not important.

Reply:
Please remember these documentations are mainly for external users, and they use DATATYPE as an enum. However I do not feel strongly about so if you also want to document the internals then please...

Reply:
Hmm, let me think about this.  To me the important part for users to understand is that datatypes aren't always 1 byte when they become registers.  For instance:
https://github.com/pymodbus-dev/pymodbus/issues/634#issuecomment-845216074
https://github.com/pymodbus-dev/pymodbus/issues/1033#issuecomment-1212168222
https://github.com/pymodbus-dev/pymodbus/discussions/1303#discussioncomment-4794593

Reply:
actually most users do not think in bits or bytes....they think 1 register == 1 float/int/string.

I have that problem constantly in home assistant and quite frequently here. 

I am open to ideas.